packages/kbn-test/src/functional_tests/cli/run_tests_cli.js

```diff
@@ -47,9 +47,13 @@ function processArgs(defaultConfigPaths) {
   const log = createToolingLog(pickLevelFromFlags(options));
   log.pipe(process.stdout);
 
+  // Default is 'snapshot', so unless option passed is 'source', use 'snapshot'.
+  const esFrom = options['es-from'] === 'source' ? 'source' : 'snapshot';
```

Comment:
Why not just use `options['es-from'] || 'snapshot'`? This ternary will allow passing `--es-from=foo` and interpret it as `--es-from=snapshot` because any none "source" value is ignored.